### PR TITLE
Testing framework to validate controller

### PR DESCRIPTION
**Pull request by cooktheryan**
Build image, deploy kind, load into kind, and run kuttl tests

### CODE REVIEW

**Comment by cooktheryan** [comment]
Example run https://github.com/cooktheryan/argocd-terraform-controller/actions/runs/2264709268

File: .github/workflows/validate-terraform-argo.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          docker inspect ${CONTROLLER_IMAGE}
+
+      - name: Login to the registry
+        run: |
+          [[ "${CONTROLLER_IMAGE}" =~ ^([^/]+)/[^/]+/[^/]+ ]] && REGISTRY="${BASH_REMATCH[1]}" || REGISTRY="quay.io"
+          echo "Attempting docker login to: ${REGISTRY}"
+          echo "${{ secrets.REGISTRY_PASSWORD }}" | docker login -u "${{ secrets.REGISTRY_USERNAME }}" --password-stdin ${REGISTRY}
+   
+      - name: Push to registry (latest)
+        run: docker push "${CONTROLLER_IMAGE}"
```
Comment (sabre1041) [inline .github/workflows/validate-terraform-argo.yaml]: If we are pushing tags I would assume that we would also want to push the tag as well?

The [docker/build-push-action](https://github.com/docker/build-push-action) supports providing a list of tags that should be made as an outcome

Comment (cooktheryan) [reply .github/workflows/validate-terraform-argo.yaml]: switched the step to use the action above. Currently we are only using latest. I think once we get a v0.0.1 identified we can add in that line

**Review by sabre1041 — CHANGES_REQUESTED** [review]
@cooktheryan When transitioning to the [build-push-action](https://github.com/docker/build-push-action), there are a few pieces of setup that needs to be performed. [Here](https://github.com/redhat-cop/github-workflows-operators/blob/main/.github/workflows/release-operator.yml#L282-L286) is an example of what would need to be added

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**